src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -270,6 +270,14 @@ public virtual void Contextualize(ViewContext viewContext)
         /// <inheritdoc />
         public IHtmlContent AntiForgeryToken()
         {
+            // If we're inside a BeginForm/BeginRouteForm, the antiforgery token might have already been
+            // created and appended to the 'end form' content.
+            if (ViewContext.FormContext.HasAntiforgeryToken)
+            {
+                return HtmlString.Empty;
+            }
```

Comment:
I don't know of any cases. If we decided to give precedence to `Html.AntiForgeryToken()` - that would complicate and code and require extra state tracking on `FormContext`.